Mathlib/Data/Real/ENNReal.lean

```diff
@@ -167,381 +180,296 @@ theorem ofReal_toReal {a : ℝ≥0∞} (h : a ≠ ∞) : ENNReal.ofReal a.toReal
 
-@[simp]
-theorem ofReal_lt_top {r : ℝ} : ENNReal.ofReal r < ∞ :=
-  lt_top_iff_ne_top.2 ofReal_ne_top
+@[simp] theorem ofReal_lt_top {r : ℝ} : ENNReal.ofReal r < ∞ := coe_lt_top
 #align ennreal.of_real_lt_top ENNReal.ofReal_lt_top
 
-@[simp]
-theorem top_ne_ofReal {r : ℝ} : ∞ ≠ ENNReal.ofReal r := by simp [ENNReal.ofReal]
+@[simp] theorem top_ne_ofReal {r : ℝ} : ∞ ≠ ENNReal.ofReal r := top_ne_coe
 #align ennreal.top_ne_of_real ENNReal.top_ne_ofReal
 
-@[simp]
-theorem zero_ne_top : 0 ≠ ∞ :=
-  coe_ne_top
+@[simp] theorem zero_ne_top : 0 ≠ ∞ := coe_ne_top
 #align ennreal.zero_ne_top ENNReal.zero_ne_top
 
-@[simp]
-theorem top_ne_zero : ∞ ≠ 0 :=
-  top_ne_coe
+@[simp] theorem top_ne_zero : ∞ ≠ 0 := top_ne_coe
 #align ennreal.top_ne_zero ENNReal.top_ne_zero
 
-@[simp]
-theorem one_ne_top : 1 ≠ ∞ :=
-  coe_ne_top
+@[simp] theorem one_ne_top : 1 ≠ ∞ := coe_ne_top
 #align ennreal.one_ne_top ENNReal.one_ne_top
 
-@[simp]
-theorem top_ne_one : ∞ ≠ 1 :=
-  top_ne_coe
+@[simp] theorem top_ne_one : ∞ ≠ 1 := top_ne_coe
 #align ennreal.top_ne_one ENNReal.top_ne_one
 
-@[simp, norm_cast]
-theorem coe_eq_coe : (↑r : ℝ≥0∞) = ↑q ↔ r = q :=
-  WithTop.coe_eq_coe
+@[simp, norm_cast] theorem coe_eq_coe : (↑r : ℝ≥0∞) = ↑q ↔ r = q := WithTop.coe_eq_coe
 #align ennreal.coe_eq_coe ENNReal.coe_eq_coe
 
-@[simp, norm_cast]
-theorem coe_le_coe : (↑r : ℝ≥0∞) ≤ ↑q ↔ r ≤ q :=
-  WithTop.coe_le_coe
+@[simp, norm_cast] theorem coe_le_coe : (↑r : ℝ≥0∞) ≤ ↑q ↔ r ≤ q := WithTop.coe_le_coe
 #align ennreal.coe_le_coe ENNReal.coe_le_coe
 
-@[simp, norm_cast]
-theorem coe_lt_coe : (↑r : ℝ≥0∞) < ↑q ↔ r < q :=
-  WithTop.coe_lt_coe
+@[simp, norm_cast] theorem coe_lt_coe : (↑r : ℝ≥0∞) < ↑q ↔ r < q := WithTop.coe_lt_coe
 #align ennreal.coe_lt_coe ENNReal.coe_lt_coe
 
-theorem coe_mono : Monotone (coe : ℝ≥0 → ℝ≥0∞) := fun _ _ => coe_le_coe.2
+theorem coe_mono : Monotone some := fun _ _ => coe_le_coe.2
 #align ennreal.coe_mono ENNReal.coe_mono
 
-@[simp, norm_cast]
-theorem coe_eq_zero : (↑r : ℝ≥0∞) = 0 ↔ r = 0 :=
-  coe_eq_coe
+@[simp, norm_cast] theorem coe_eq_zero : (↑r : ℝ≥0∞) = 0 ↔ r = 0 := coe_eq_coe
 #align ennreal.coe_eq_zero ENNReal.coe_eq_zero
 
-@[simp, norm_cast]
-theorem zero_eq_coe : 0 = (↑r : ℝ≥0∞) ↔ 0 = r :=
-  coe_eq_coe
+@[simp, norm_cast] theorem zero_eq_coe : 0 = (↑r : ℝ≥0∞) ↔ 0 = r := coe_eq_coe
 #align ennreal.zero_eq_coe ENNReal.zero_eq_coe
 
-@[simp, norm_cast]
-theorem coe_eq_one : (↑r : ℝ≥0∞) = 1 ↔ r = 1 :=
-  coe_eq_coe
+@[simp, norm_cast] theorem coe_eq_one : (↑r : ℝ≥0∞) = 1 ↔ r = 1 := coe_eq_coe
 #align ennreal.coe_eq_one ENNReal.coe_eq_one
 
-@[simp, norm_cast]
-theorem one_eq_coe : 1 = (↑r : ℝ≥0∞) ↔ 1 = r :=
-  coe_eq_coe
+@[simp, norm_cast] theorem one_eq_coe : 1 = (↑r : ℝ≥0∞) ↔ 1 = r := coe_eq_coe
 #align ennreal.one_eq_coe ENNReal.one_eq_coe
 
-@[simp]
-theorem coe_nonneg : 0 ≤ (↑r : ℝ≥0∞) :=
-  coe_le_coe.2 <| zero_le _
+-- porting note: no longer a `simp`, deprecated, protected
+@[deprecated] protected theorem coe_nonneg : 0 ≤ (r : ℝ≥0∞) := zero_le _
```

Comment:
Please give an argument to `deprecated` that informs users what they should use instead.